### PR TITLE
chore: adds go mod tidy and go work sync for all modules.

### PR DESCRIPTION
**Pull request by jcchavezs**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -69,15 +71,43 @@ func Lint() error {
...
+		if err != nil {
+			return err
+		}
 
-	if err := sh.RunV("go", "work", "sync"); err != nil {
+		if d.IsDir() {
+			return nil
+		}
+
+		runGoModTidy := strings.HasSuffix(path, "go.mod")
```
Comment (anuraaga) [inline magefile.go:83]: Would use `path.Base` to be more precise

File: magefile.go [hunk 2]
```diff
@@ -69,15 +71,43 @@ func Lint() error {
...
+			out, err := cmd.Output()
+			fmt.Printf(string(out))
+			if err != nil {
+				return err
+			}
+		}
+
+		runGoWorkSync := strings.HasSuffix(path, "go.work") || runGoModTidy
+		if runGoWorkSync {
+			cmd := exec.Command("go", "work", "sync")
```
Comment (anuraaga) [inline magefile.go:96]: I don't think go work sync is supposed to be run in subdirectories (it probably does work, but it's meant to be a global command at top level). I have found that it seems to generally need to be run before go mod tidy too

**Comment by jcchavezs** [comment]
PTAL @anuraaga

File: magefile.go [hunk 3]
```diff
@@ -69,15 +70,41 @@ func Lint() error {
...
+		if err != nil {
+			return err
+		}
 
-	if err := sh.RunV("go", "work", "sync"); err != nil {
+		if !d.IsDir() {
+			return nil
+		}
+
+		if _, err := os.Stat(filepath.Join(path, "go.work")); err == nil {
```
Comment (anuraaga) [inline magefile.go:82]: I think this can be outside the walkdir itself

Comment (jcchavezs) [reply magefile.go]: Makes sense.

**Comment by jcchavezs** [comment]
@anuraaga PTAL

**Review by anuraaga — APPROVED** [review]

**Pull request merged**